tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc

```diff
@@ -76,7 +76,83 @@ std::set<string> GetDeletedServables(
   return deleted_servables;
 }
 
-// Like PollFileSystemForConfig(), but for a single servable.
+//Add a new ServableData for the model version to the the vector of versions to aspired
+void UpdateVersionToAspire(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    const string version_relative_path,
+    const int version_number,
+    std::vector<ServableData<StoragePath>>* versions) {
+    const ServableId servable_id = {servable.servable_name(), version_number};
+    const string full_path =
+        io::JoinPath(servable.base_path(), version_relative_path);
+    versions->emplace_back(ServableData<StoragePath>(servable_id, full_path));
+}
+
+bool IsValidVersionNumber(const string& version_path, int64 *version_number) {
+  return strings::safe_strto64(version_path.c_str(), version_number);
+}
+
+// Update the servable data to include
+// all the model versions found in the base path as aspired versions.
+bool EmitAllVersions(
+    const FileSystemStoragePathSourceConfig::ServableToMonitor& servable,
+    const std::vector<string>& children,   // represents a list of base-path children from the file system.
+    std::vector<ServableData<StoragePath>>* versions) {
+
+  LOG(INFO) << "Emit all servable versions";
+
+  bool version_found = false;
+  for (int i = 0; i < children.size(); ++i) {
```

Comment:
Please drop the 'i', and instead write:
for (const string& child : children) {
  ... child ...